### ISSUE TITLE
then callback?

### ISSUE THREAD

**Issue by yohanboniface** [Leaflet/prosthetic-hand#6]
When using prosthetic-hand for unittesting, it would be great to have a way of setting a final callback function where to do assert and expect checks.

According to the example I see on Leaflet tests, the current pattern is to add a setTimeout for this.

What about introducing a Finger `then` method to register callbacks?

API would look like:

```
mouse.wait(100).moveTo(200, 200, 0).down().moveBy(5, 0, 20).moveBy(256, 32, 200).up().then(function () {
    assert(everythingIsGood);
});
```

**Comment by jieter** [comment]
Yes,was thinking of that too,  would be really handy

**Comment by IvanSanchez** [comment]
I'm against naming anything `then`, because that can cause confusion when dealing with `Promise`s.

The `Hand` already fires a DOM event into the `document` when it's finished (I haven't updated the documentation yet), but it's flimsy - PhantomJS doesn't like it for some reason, and it messes up the context and the variable scope.

I'm wondering whether to copy-paste some code from Leaflet's `Evented` and have things like `Hand.on()`, `Hand.once()` and `Hand.fire()`.

**Comment by jieter** [comment]
Are you only opposed to the name `then`, or to the idea of callback in general? I think a reliable way to mark the end of the queue is a must have, as is demonstrated by the failing leaflet builds.

**Comment by IvanSanchez** [comment]
@jieter Just to the `then` name in particular. I agree a more robust way of doing callbacks is needed.

**Comment by IvanSanchez** [comment]
I think this is covered by e09776d8f601191950c38ffeeb12756db6f2fb71, released in 1.3.0. Reopen if that doesn't cover your use cases.
